### ISSUE TITLE
Feature request: Source for RPC stored in metadata

### ISSUE THREAD

**Issue by aviborg** [OSGeo/gdal#6323]
When reading a .TIF or .NTF (those I've tested) that contains either a separate RPC-file or has RPC-data embedded, the source of the data is not stored in the RPC-metadata of the GDAL dataset. For example if there is a .RPB, RPC.TXT and a .XML stored together with the .TIF, then the .RPB will be used base on some some prioritization rules from such as: https://gdal.org/drivers/raster/gtiff.html#georeferencing 

But there is no way to tell from where the RPC-metadata originated by querying the GDALDataset metadata.

An enhancement would be to add a RPC-metadata entry with the source of the RPC-data.

**Comment by BishopGIS** [comment]
It seems to me RPC already is in metadata. See https://gdal.org/user/raster_data_model.html#rpc-domain

**Comment by aviborg** [comment]
> It seems to me RPC already is in metadata. See https://gdal.org/user/raster_data_model.html#rpc-domain

No, that is just the RPC data. There is no way to tell the source of this RPC metadata. I was more thinking like:

```
Metadata RPC:
      URL: /path/to/folder/file.RPB
      ERR_BIAS: 2.3456
      .... 
```

**Comment by jratike80** [comment]
Gdalinfo does report something but I guess that this is not what you want:

```
gdalinfo  md_kompsat.tif
Driver: GTiff/GeoTIFF
ERROR 1: Maximum number of characters allowed reached.
ERROR 1: md_kompsat.rpc file found, but missing LINE_OFF field (and possibly others).
Files: md_kompsat.tif
       md_kompsat.txt
       md_kompsat.rpc
```

**Comment by aviborg** [comment]
> Gdalinfo does report something but I guess that this is not what you want:
> 
> ```
> gdalinfo  md_kompsat.tif
> Driver: GTiff/GeoTIFF
> ERROR 1: Maximum number of characters allowed reached.
> ERROR 1: md_kompsat.rpc file found, but missing LINE_OFF field (and possibly others).
> Files: md_kompsat.tif
>        md_kompsat.txt
>        md_kompsat.rpc
> ```

You can use `GDALGetFileList` but if you have a .RPB and .XML file both will be listed and you can't tell which one that was used to populate the metadata struct.

**Comment by rouault** [comment]
Storing the source in the RPC metadata domain could cause inconveniences when it is copied to other formats or other operations where we don't really want to keep that information (irrelevant, or privacy issues)
Perhaps have a PROVENANCE metadata domain with keys like GEOTRANSFORM, CRS, GCP, RPC, ... ?

**Comment by jratike80** [comment]
There is some documentation in https://gdal.org/drivers/raster/gtiff.html, maybe it could be improved?

> GDAL can read and write the RPCCoefficientTag as described in the [RPCs in GeoTIFF](http://geotiff.maptools.org/rpc_prop.html) proposed extension. The tag is written only for files created with the default profile GDALGeoTIFF. For other profiles, a .RPB file is created. In GDAL data model, the RPC coefficients are stored into the RPC metadata domain. For more details, see the [RFC 22: RPC Georeferencing](https://gdal.org/development/rfc/rfc22_rpc.html#rfc-22). If .RPB or _RPC.TXT files are found, they will be used to read the RPCs, even if the RPCCoefficientTag tag is set.

Something similar happens with other kind of GeoTIFFs
> By default, information is fetched in following order (first listed is the most prioritary): PAM (Persistent Auxiliary metadata) .aux.xml sidecar file, INTERNAL (GeoTIFF keys and tags), TABFILE (.tab), WORLDFILE (.tfw, .tifw/.tiffw or .wld).

But isn't that sort of runtime metadata that cannot be saved permanently? Other software may behave in a different way.

**Comment by aviborg** [comment]
> Storing the source in the RPC metadata domain could cause inconveniences when it is copied to other formats or other operations where we don't really want to keep that information (irrelevant, or privacy issues) Perhaps have a PROVENANCE metadata domain with keys like GEOTRANSFORM, CRS, GCP, RPC, ... ?

Yes, any other domain would do.
